### ISSUE TITLE
Custom network range when using --network-address

### ISSUE THREAD

**Issue by splazit** [abiosoft/colima#573]
### Description

I'm not sure if this feature is already available or how to do it, but what I would like to do is when creating a new Colima vm with --network-address flag, I would like to customize the network for that. Right now it seems it defaults to 192.168.106.x.
My use case it that when I connected to my company VPN, there's a certain subnet networks I can configure so that traffic won't go to the VPN, but the 192.168.106.x is not in the list and I cannot access the Colima vm when I'm on VPN.

**Comment by jandubois** [comment]
Unless colima does something special, the subnets are configured in Lima via the `~/.lima/_config/networks.yaml` file. You should stop all your VMs before editing this file.

**Comment by splazit** [comment]
Thanks for prompt response.

Yeah, I’ve tried that but it does not take effective, I also tried to update the /etc/sudoers.d/colima.

It seems to me that it’s hard coded in here: https://github.com/abiosoft/colima/blob/main/daemon/process/vmnet/vmnet.go#L23

I could be wrong though.

From: Abiola Ibrahim ***@***.***>
Date: Thursday, January 5, 2023 at 12:38 PM
To: abiosoft/colima ***@***.***>
Cc: Hieu Nguyen ***@***.***>, Author ***@***.***>
Subject: Re: [abiosoft/colima] Custom network range when using --network-address (Issue #573)

Unless colima does something special, the subnets are configured in Lima via the ~/.lima/_config/networks.yaml file. You should stop all your VMs before editing this file.

Nothing too special other than choosing 192.160.106.x range instead of the default 192.168.105.x.

It is possible to include the ability for users to specify a custom range.
However, considering that the new VZ driver do not (yet) have that capability, I am a bit hesitant to welcome the idea.

**Comment by abiosoft** [comment]
> It seems to me that it’s hard coded in here: https://github.com/abiosoft/colima/blob/main/daemon/process/vmnet/vmnet.go#L23

@splazit yeah it is. You need to change that line and the next, then you'd also need to modify https://github.com/abiosoft/colima/blob/main/embedded/network/sudo.txt accordingly.

After that, `make && make install` and you should be good to go.

**Comment by splazit** [comment]
thanks for the pointer. Would it possible to have this configurable in `~/.colima/_templates/default `or  use the values from `~/.lima/_config/networks.yaml` as mentioned above?

**Comment by splazit** [comment]
tried to compile a new binary but unfortunately it does not work as I can't even create a new vm.

**Comment by abiosoft** [comment]
> tried to compile a new binary but unfortunately it does not work as I can't even create a new vm.

can you share the error you are getting?

**Comment by splazit** [comment]
I use this command w/ the new compiled colima:
colima start -a x86_64 -c 6 -d 120 -m 8 --vm-type qemu --network-address

Console output: 
`
INFO[0000] starting colima
INFO[0000] runtime: docker
INFO[0000] preparing network ...                         context=vm
WARN[0015] error starting network: error at 'preparing network': error running [/opt/local/bin/colima daemon status default], output: "time=\"2023-01-05T22:53:13-07:00\" level=fatal msg=\"pid file not found: stat /Users/xxxxxx/.colima/default/daemon/daemon.pid: no such file or directory\"", err: "exit status 1"  context=vm
INFO[0015] creating and starting ...                     context=vm
WARN[0015] error setting up reachable IP address: vmnet socket file not found: stat /Users/xxxxxxx/.colima/default/daemon/vmnet.sock: no such file or directory
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"
> [hostagent] Waiting for the essential requirement 1 of 5: "ssh"`

**Comment by Orgil** [comment]
@abiosoft exact same error i get when i tried to compile. please make it possible to configurable.

**Comment by katzelad** [comment]
@abiosoft Was there any progress with this?
I also need to use Colima with Testcontainers alongside a corporate VPN and this is a major blocker preventing me to do so. Is there any workaround besides compilation from source?
Thanks

**Comment by splazit** [comment]
Btw, I no longer need to use Colima and opted for using Lima with the Rancher ISO image. It’s lighter and been working quite well for me.

From: Elad ***@***.***>
Date: Monday, July 22, 2024 at 8:32 AM
To: abiosoft/colima ***@***.***>
Cc: Hieu Nguyen ***@***.***>, Mention ***@***.***>
Subject: Re: [abiosoft/colima] Custom network range when using --network-address (Issue #573)

@abiosoft<https://github.com/abiosoft> Was there any progress with this?
I also need to use Colima with Testcontainers alongside a corporate VPN and this is a major blocker preventing me to do so. Is there any workaround other than compilation from source?
Thanks
